app/models/concerns/orchestration/compute.rb

```diff
@@ -290,9 +290,13 @@ def match_macs_to_nics(fog_attr)
     # array so we can delete from it safely
     fog_nics = vm.interfaces.dup
 
+    logger.debug "Orchestration::Compute: Trying to match network interfaces from fog #{fog_nics.inspect}"
     self.interfaces.each do |nic|
       selected_nic = vm.select_nic(fog_nics, nic)
-      next if selected_nic.nil? # found no matching fog nic for this Foreman nic, move on
+      if selected_nic.nil? # found no matching fog nic for this Foreman nic
+        logger.warn "Orchestration::Compute: Could not match network interface #{nic.inspect}"
+        return failure(_("Could not find virtual machine network interface matching %s") % (nic.identifier || nic.ip || nic.name || nic.type))
```

Comment:
Sorry to say, but there's another problem with `||`, it doesn't handle the blank strings which are present when creating a host through the UI:

<pre>
2015-05-13 11:51:54 [W] Orchestration::Compute: Could not match network interface #<Nic::Managed id: nil, mac: nil, ip: "192.168.122.111", type: "Nic::Managed", name: "debug5.example.com", host_id: nil, subnet_id: 4, domain_id
: 6, attrs: {}, created_at: nil, updated_at: nil, provider: nil, username: nil, password: nil, virtual: false, link: true, identifier: "", tag: "", attached_to: "", managed: true, mode: "balance-rr", attached_devices: "", bond
_options: "", primary: true, provision: true, compute_attributes: {"type"=>"network", "network"=>"default", "model"=>"virtio"}>
2015-05-13 11:51:54 [W] »Could not find virtual machine network interface matching «
</pre>

Perhaps `[nic.identifier, nic.ip, ...].find?(&:present?)` or something?